ang/prospect/services/convert-prospect-case-action.service.js

```diff
@@ -0,0 +1,79 @@
+(function (angular, $, _) {
+  var module = angular.module('prospect');
+
+  module.service('ConvertProspectCaseAction', ConvertProspectCaseAction);
+
+  function ConvertProspectCaseAction ($location, crmApi, ProspectGlobalValues) {
+    var isConvertedToProspect = false;
+
+    this.refreshData = checkIfConvertedToProspect;
+
+    /**
+     * Checks if the Action is allowed
+     */
+    this.isActionAllowed = function () {
+      return checkIfProspectingCaseTypeCategory() && !isConvertedToProspect;
+    };
+
+    /**
+     * Click event handler for the Action
+     *
+     * @param {Array} cases
+     * @param {Object} action
+     * @param {Function} callbackFn
+     */
+    this.doAction = function (cases, action, callbackFn) {
+      if (action.type !== 'contribution' && action.type !== 'pledge') {
```

Comment:
```js
if (_.includes(['contribution', 'pledge'], action.type)) {
  // ...
}
```

Also, why is this here and not on `isActionAllowed` ? when I run a method I expect that method to work and not exit silently with a `return;`.

Reply:
It was copied from existing code. Fixed.